API.md

```diff
@@ -1,4 +1,111 @@
-# Authmosphere API documentation
+# Authmosphere API Documentation
+
+## OAuth Tooling
+
+This tooling provides helper functions to request and validate tokens based on the OAuth 2.0 [RFC 6749](https://tools.ietf.org/html/rfc6749) specification.
+
+### getAccessToken
+
+Requests a token based on the given configuration (which specifies the grant type and corresponding parameters). See the [`OAuthConfig` documentation](#tbd) for details.
+
+#### Usage
+
+```ts
+import {
+  OAuthGrantType,
+  ClientCredentialsGrantConfig,
+  Token,
+  getAccessToken
+} from 'authmosphere';
+
+const config: ClientCredentialsGrantConfig = {
+  grantType: OAuthGrantType.CLIENT_CREDENTIALS_GRANT,
+  credentialsDir: './crendentials',
+  accessTokenEndpoint: 'example.com/token',
+  scopes: ['my-app.read', 'my-app.write'];
+};
+
+getAccessToken(config)
+  .then((token: Token) => {
+    // ...use the token...
+  })
+  .catch((err) => {
+    // ...handle the error...
+  });
+```
+
+#### Signature
+
+`getAccessToken(config[, logger]): Promise<Token>`
+
+#### Arguments
+
+* [`config: OAuthConfig`](#tbd) - OAuth configuration for the request (specify grant type and corresponding parameters)
+* [`logger?: Logger`](#tbd) - Optional logger
+
+#### Returns
+
+`Promise<Token>`
```

Comment:
The Signature already describe this (we handled that also in mocktooling with the signature).
Section would make sense if we had to add some prose.

Reply:
👍  I will add some prosa